etc/kayobe/ansible/get-cloud-facts.yml

```
@@ -0,0 +1,124 @@
+          distributions: "{{ hostvars | group_hostvars_by_var('distribution') }}"
+          dist_releases: "{{ hostvars | group_hostvars_by_var('distribution_release') }}"
+          kernel_versions: "{{ hostvars | group_hostvars_by_var('kernel_version','stdout') }}"
+          deployed_containers: "{{ hostvars | get_hostvars_by_host('containers_list', 'stdout_lines') | default({}) }}"
+          openstack_release: "{{ openstack_release }}"
+          openstack_release_name: "{{ openstack_release_codename }}"
+          ansible_control_host_is_vm: "{{ ansible_facts.virtualization_role == 'guest' }}"
+          controller_count: "{{ groups['controllers'] | length }}"
+          hypervisor_count: "{{ groups['hypervisors'] | length }}"
+          monitoring_count: "{{ groups['monitoring'] | length }}"
+          osd_count: "{{ groups['osds'] | length }}"
+          compute_count: "{{ groups['compute'] | length }}"
+          baremetal_count: "{{ groups['baremetal-compute'] | length }}"
+          ceph_deployed: "{{ groups['ceph'] | length > 0 | bool }}"
+          ceph_count: "{{ groups['ceph'] | length }}"
+          ceph_release: "{{ cephadm_ceph_release }}"
+          storage_hyperconverged: "{{ groups['controllers'] | intersect(groups['osds']) | length > 0 | bool }}"
+          wazuh_enabled: "{{ groups['wazuh-agent'] | length > 0 | bool }}"
+          ngs_enabled: "{{ ngs_enabled | default(false) }}"
+          kayobe_managed_switches: "{{ groups['switches'] | length > 0 | bool }}"
+          proxy_configured: "{{ http_proxy | bool or https_proxy | bool }}"
+          bifrost_version: "{{ kolla_bifrost_source_version }}"
+          internal_tls_enabled: "{{ kolla_enable_tls_internal }}"
+          external_tls_enabled: "{{ kolla_enable_tls_external }}"
+          firewalld_enabled_all: >-
+            {{
+              controller_firewalld_enabled and
+              compute_firewalld_enabled and
+              storage_firewalld_enabled and
+              monitoring_firewalld_enabled and
+              infra_vm_firewalld_enabled and
+              seed_firewalld_enabled and
+              seed_hypervisor_firewalld_enabled
+            }}
+          firewalld_enabled_any: >-
+            {{
+              controller_firewalld_enabled or
+              compute_firewalld_enabled or
+              storage_firewalld_enabled or
+              monitoring_firewalld_enabled or
+              infra_vm_firewalld_enabled or
+              seed_firewalld_enabled or
+              seed_hypervisor_firewalld_enabled
+            }}
+          stackhpc_package_repos_enabled: "{{ stackhpc_repos_enabled }}"
+          pulp_tls_enabled: "{{ pulp_enable_tls }}"
+          kolla_image_tags: "{{ kolla_image_tags }}"
+          gpu_passthrough_enabled: "{{ gpu_group_map | length > 0 | bool }}"
+          vGPU_enabled: "{{ groups['vgpu'] | length > 0 | bool }}"
```

Just by convention, lets keep all vars lowercase
```suggestion
          vgpu_enabled: "{{ groups['vgpu'] | length > 0 | bool }}"
```